main/views.py

```diff
@@ -81,12 +44,17 @@ def api_players(request):
         data = [{"id": player.id, "name": player.name} for player in players]
         return JsonResponse({"players": data})
     else:
-        data = json.loads(request.body)
-        player = Player.objects.create(name=data["name"])
-        return JsonResponse({
-            "id": player.id,
-            "name": player.name
-        })
+        try:
+            data = json.loads(request.body)
+            player = Player.objects.create(name=data["name"])
+            return JsonResponse({
+                "id": player.id,
+                "name": player.name
+            })
+        except ValueError:
+            data = request.POST
+            Player.objects.create(name=data["name"])
+            return redirect('players')
```

Comment:
I see how the task I gave you was not optimal here - we don't usually call the same endpoint with different data types (eg. json/form data). I think here it would also be a lot nicer if you created two different views - one for the api, one for the form.